src/runtime/mod.rs

```diff
@@ -364,13 +364,13 @@ impl Module {
         // NB: It is strange that there is not a way to do this already. The Display impl
         // seems to only provide the path.
         let uri = req.uri.clone();
-        let authority = format!("{}:{}", host, port);
         headers.insert(
             "X_FULL_URL".to_owned(),
             format!(
-                "{}://{}{}",
+                "{}://{}:{}{}",
                 uri.scheme_str().unwrap_or("http"), // It is not clear if Hyper ever sets scheme.
```

Comment:
With https, does this need to be handled differently at all?

Reply:
Only if Hyper does not set scheme to `https`.